jazz_metrics/components/utils.js

```diff
@@ -0,0 +1,224 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+    Helper functions for Metrics
+    @module: utils.js
+    @description: Defines functions like format the output as per metrics catalog.
+    @author:
+    @version: 1.0
+**/
+const parser = require('aws-arn-parser');
+const metricConfig = require("./metrics.json");
+const global_config = require("../config/global-config.json");
+
+function massageData(assetResults, eventBody) {
+
+    var output_obj = {};
+    output_obj = {
+        "domain": eventBody.domain,
+        "service": eventBody.service,
+        "environment": eventBody.environment,
+        "end_time": eventBody.end_time,
+        "start_time": eventBody.start_time,
+        "interval": eventBody.interval,
+        "statistics": eventBody.statistics,
+        "assets": assetResults
+    }
+
+    return output_obj;
+};
+
+function assetData (results, assetItem) {
+    var asset_obj = {};
+   
+    asset_obj = {
+        "type": assetItem.type,
+        "asset_name": assetItem.asset_name,
+        "statistics": assetItem.statistics,
+        "metrics": []
+    };
+
+    var metricsArr = [];
+
+    results.forEach(function(key) {
+        var metricsObj = {
+            "metric_name": key.Label,
+            "datapoints": key.Datapoints
+        }
+        metricsArr.push(metricsObj);
+    });
+    asset_obj.metrics = metricsArr;
+    return asset_obj;
+};
+
+function getNameSpaceAndMetriDimensons (nameSpaceFrmAsset){
+    var output_obj = {};
+    output_obj["isError"] = false;
+    var paramMetrics = [];
+    var nameSpace = nameSpaceFrmAsset.toLowerCase();
+    
+    var namespacesList = metricConfig.namespaces;
+
+    var supportedNamespace = namespacesList[nameSpace];
+
+    paramMetrics = supportedNamespace["metrics"];
+
+    var awsAddedNameSpace = nameSpace.indexOf('aws/') === -1 ? 'aws/' + nameSpace : nameSpace;
+    awsAddedNameSpace = awsAddedNameSpace.replace(/ /g, "");
+    var awsNameSpace;
+
+    switch (awsAddedNameSpace) {
```

Comment:
why not use an object here
awsNameSpace = {key: value} this way it will be more concise and better.

Reply:
mapping is done for Namespace